document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt

```diff
@@ -29,80 +29,54 @@ import com.ritense.smartdocuments.dto.SmartDocumentsPropertiesDto
-
 
 class SmartDocumentsClient(
     private var smartDocumentsConnectorProperties: SmartDocumentsConnectorProperties,
-    private val smartDocumentsWebClientBuilder: WebClient.Builder,
+    private val smartDocumentsRestClientBuilder: RestClient.Builder,
     private val maxFileSizeMb: Int,
     private val temporaryResourceStorageService: TemporaryResourceStorageService,
 ) {
 
     fun getSmartDocumentsTemplateData(smartDocumentsPropertiesDto: SmartDocumentsPropertiesDto): SmartDocumentsTemplateData? {
-        return pluginWebClient(smartDocumentsPropertiesDto).get()
+        val response = pluginRestClient(smartDocumentsPropertiesDto)
+            .get()
             .uri(STRUCTURE_PATH)
             .retrieve()
-            .bodyToMono(String::class.java)
-            .map { xmlData -> xmlMapper.readValue(xmlData, SmartDocumentsTemplateData::class.java) }
-            .doOnError { throw toHttpClientErrorException(it) }
-            .block()
+            .body<String>()!!
+        return xmlMapper.readValue(response, SmartDocumentsTemplateData::class.java)
     }
 
-    fun generateDocument(
-        smartDocumentsRequest: SmartDocumentsRequest,
-    ): FilesResponse {
-        return webClient().post()
+    fun generateDocument(smartDocumentsRequest: SmartDocumentsRequest): FilesResponse {
+        return restClient()
+            .post()
             .uri("/wsxmldeposit/deposit/unattended")
             .contentType(APPLICATION_JSON_UTF8)
-            .bodyValue(fixRequest(smartDocumentsRequest))
+            .body(fixRequest(smartDocumentsRequest))
             .retrieve()
-            .bodyToMono(FilesResponse::class.java)
-            .doOnError { throw toHttpClientErrorException(it) }
-            .block()!!
+            .body<FilesResponse>()!!
     }
 
     fun generateDocumentStream(
         smartDocumentsRequest: SmartDocumentsRequest,
         outputFormat: DocumentFormatOption,
     ): FileStreamResponse {
-        val response = webClient().post()
+        val result = restClient()
+            .post()
             .uri("/wsxmldeposit/deposit/unattended")
             .contentType(APPLICATION_JSON_UTF8)
-            .bodyValue(fixRequest(smartDocumentsRequest))
-            .exchange()
-            .block()
-
-        val responseOut = PipedOutputStream()
-        val responseIn = PipedInputStream(responseOut)
-        val body = response.body(BodyExtractors.toDataBuffers())
-            .doOnError {
-                responseIn.use {  }
-                throw toHttpClientErrorException(it)
-            }
-            .doFinally { responseOut.use {  } }
-
-        DataBufferUtils.write(body, responseOut).subscribe(DataBufferUtils.releaseConsumer())
-
-        assertHttp200Status(HttpStatus.valueOf(response.statusCode().value()), response, responseIn)
+            .body(fixRequest(smartDocumentsRequest))
+            .retrieve()
+            .body<ByteArray>()!!
```

Comment:
By responding with a ByteArray, the entire document is loaded into the RAM which is not be possible for larger documents.

This `generateDocumentStream(..)` was made to replace the old `generateDocument(..)` function which did everything in-memory.